cmd/juju/application/bind.go

```diff
@@ -0,0 +1,238 @@
+	Get(string, string) (*params.ApplicationGetResults, error)
+	MergeBindings(req params.ApplicationMergeBindingsArgs) error
+}
+
+// Bind is responsible for changing the bindings for an application.
+type bindCommand struct {
+	modelcmd.ModelCommandBase
+
+	NewApplicationClient func(base.APICallCloser) ApplicationBindClient
+	NewModelConfigGetter func(base.APICallCloser) ModelConfigGetter
+	NewSpacesClient      func(base.APICallCloser) SpacesAPI
+
+	ApplicationName string
+	BindExpression  string
+	Bindings        map[string]string
+	Force           bool
+}
+
+const bindCmdDoc = `
+When no bind expression is specified, the command will print out the currently
+assigned bindings for the application.
+
+To update the default binding for the application and automatically update all
+existing endpoint bindings that were referencing the old default, you can use 
+the following syntax:
+
+  juju bind foo new-default
+
+To bind individual endpoints to a space you can use the following syntax:
+
+  juju bind foo endpoint-1=space-1 endpoint-2=space-2
+
+Finally, the above commands can be combined to update both the default space
+and individual endpoints in one go:
+
+  juju bind foo new-default endpoint-1=space-1
+
+In order to be able to bind any endpoint to a space, all machines where the
+application units are deployed to are required to be configured with an address
+in that space. However, you can use the --force option to bypass this check.
+`
+
+func (c *bindCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:    "bind",
+		Args:    "<application> [<default-space>] [<endpoint-name>=<space> ...]",
+		Purpose: "Change bindings for a deployed application.",
+		Doc:     bindCmdDoc,
+	})
+}
+
+func (c *bindCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.Force, "force", false, "Allow endpoints to be bound to spaces that might not be available to all existing units")
+}
+
+func (c *bindCommand) Init(args []string) error {
+	nArgs := len(args)
+	if nArgs == 0 {
+		return errors.Errorf("no application specified")
+	}
+
+	if !names.IsValidApplication(args[0]) {
+		return errors.Errorf("invalid application name %q", args[0])
+	}
+	c.ApplicationName = args[0]
+	c.BindExpression = strings.Join(args[1:], " ")
+	return nil
+}
+
+// Run connects to the specified environment and applies the requested binding
+// changes.
+func (c *bindCommand) Run(ctx *cmd.Context) error {
+	apiRoot, err := c.NewAPIRoot()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer func() { _ = apiRoot.Close() }()
+
+	if err = c.checkApplicationFacadeSupport(apiRoot, "changing application bindings", 11); err != nil {
+		return err
+	}
+
+	if err = c.parseBindExpression(apiRoot); err != nil {
+		return err
+	}
+
+	modelConfigGetter := c.NewModelConfigGetter(apiRoot)
+	modelConfig, err := getModelConfig(modelConfigGetter)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	generation, err := c.ActiveBranch()
```

Comment:
Although it doesn't matter, because bindings are not branch-based, I think we should skip this and pass `GenerationMaster` to the `Get` call.

Reply:
This is written as such to mirror the behavior of the upgrade-charm command for consistency purposes. Can we leave it as-is? No objection to changing it though. Thoughts?